### PR TITLE
Create Input Control

### LINKED ISSUE #3: Input | Create the basic input control
Doc:
https://modus.trimble.com/components/mobile/inputs/

All the properties of the input should be exposed to the InputControl

### PR DESCRIPTION
**Pull request by BAL4-R**
Resolves #3, #4, #5, #6, #56

Create basic input.

Add left and right icons in the input control.

Add different styles of input.

Shows helper text.

Allows user to display success or error status below the input based on the input validation function.

Screenshot_20230511-140827__01
Screenshot_20230511-140854
Screenshot_20230511-140822
Screenshot_20230511-140836

### CODE REVIEW

File: DemoApp/DemoApp/TMInputPage.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    private void modusInput_Focused(object sender, EventArgs e)
+    {
+        Console.WriteLine("Container focused");
+
+    }
+
+    private Tuple<bool, string> CustomValidateInput(object sender)
+    {
+        var input = sender as TMInput;
+        if(input.Text.Length > 3)
```
Comment (rthanga1) [inline DemoApp/DemoApp/TMInputPage.xaml.cs:20]: This method can return null in which case the helper text should be displayed in the input.

File: Trimble.Modus.Components/Controls/TMInput/TMInput.cs [hunk 2]
```diff
@@ -0,0 +1,871 @@
...
+        /// <summary>
+        /// This method is triggered when the text in the input field is changed, 
+        /// this performs text validation to update the validation text field
+        /// </summary>
+        /// <param name="bindable">Object</param>
+        /// <param name="oldValue">Old value</param>
+        /// <param name="newValue">New value</param>
+        private static void OnInputTextChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            var tMinput = (TMInput)bindable;
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMInput/TMInput.cs:502]: Please change the variable name to `tmInput`

File: Trimble.Modus.Components/Controls/TMInput/TMInput.cs [hunk 3]
```diff
@@ -0,0 +1,871 @@
...
+            if(tMinput.InputValidation != null)
+            {
+                if (!string.IsNullOrEmpty((string)newValue))
+                {
+                    var result = tMinput.InputValidation?.Invoke(tMinput);
+
+                    tMinput._validationLabel.Text = result.Item2;
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMInput/TMInput.cs:509]: When the `result.Item2` is null or empty string, then the helper text should be displayed.

File: Trimble.Modus.Components/Controls/TMInput/TMInput.cs [hunk 4]
```diff
@@ -0,0 +1,871 @@
...
+            {
+                _border.Stroke = (Color)BaseComponent.colorsDictionary()["TrimbleBlue"];
+                _border.StrokeThickness = 2;
+                Focused?.Invoke(this, e);
+            }
+            else
+            {
+                _border.Stroke = (Color)BaseComponent.colorsDictionary()["Black"];
+                _border.StrokeThickness = 1;
+                Unfocused?.Invoke(this, e);
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMInput/TMInput.cs:772]: When unfocused, the error/success message should go away and helper text should be displayed

**Review by rthanga1 — CHANGES_REQUESTED** [review]

File: DemoApp/DemoApp/TMInputPage.xaml.cs (hunk 1 above)
Comment (BAL4-R) [reply DemoApp/DemoApp/TMInputPage.xaml.cs:20]: Done

File: Trimble.Modus.Components/Controls/TMInput/TMInput.cs (hunk 2 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMInput/TMInput.cs]: Done

File: Trimble.Modus.Components/Controls/TMInput/TMInput.cs [hunk 5]
```diff
@@ -0,0 +1,871 @@
...
+        /// <param name="oldValue">Old value</param>
+        /// <param name="newValue">New value</param>
+        private static void OnInputTextChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            var tMinput = (TMInput)bindable;
+            if(tMinput.InputValidation != null)
+            {
+                if (!string.IsNullOrEmpty((string)newValue))
+                {
+                    var result = tMinput.InputValidation?.Invoke(tMinput);
```
[the remark this answers is not in the document]
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMInput/TMInput.cs]: Done

File: Trimble.Modus.Components/Controls/TMInput/TMInput.cs (hunk 3 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMInput/TMInput.cs]: Done

File: Trimble.Modus.Components/Controls/TMInput/TMInput.cs (hunk 4 above)
Comment (BAL4-R) [reply Trimble.Modus.Components/Controls/TMInput/TMInput.cs:787]: Done

**Review by rthanga1 — APPROVED** [review]

File: Trimble.Modus.Components/Trimble.Modus.Components.csproj
Comment (ElishaSamPeterPrabhu) [inline Trimble.Modus.Components/Trimble.Modus.Components.csproj:1]: Can you remove the item group for icons being removed ? Line 20-23

Comment (BAL4-R) [reply Trimble.Modus.Components/Trimble.Modus.Components.csproj:1]: Done

**Review by ElishaSamPeterPrabhu — APPROVED** [review]

**Pull request merged**